test/spec/controllers/partyCtrlSpec.js

```diff
@@ -33,6 +36,101 @@ describe("Party Controller", function() {
     });
   });
 
+  describe('initialization', function() {
+    beforeEach(function() {
+      sandbox.stub(rootScope, 'openModal');
+    });
+
+    context('party has 1 member', function() {
+      beforeEach(function() {
+        scope.group = { memberCount: 1 };
+      });
+
+      it('awards no new achievements', function() {
+        expect(User.set).to.not.be.called;
+        expect(rootScope.openModal).to.not.be.called;
+      });
+    });
+
+    context('party has 2 members', function() {
+      beforeEach(function() {
+        scope.group = { memberCount: 2 };
+      });
+
+      context('user does not have "Party Up" achievement', function() {
+        it('awards "Party Up" achievement', function() {
+          expect(User.set).to.be.calledOnce;
+          expect(User.set).to.be.calledWith(
+            { 'achievements.partyUp': true}
+          );
+          expect(rootScope.openModal).to.be.calledOnce;
+          expect(rootScope.openModal).to.be.calledWith(
+            'achievements/partyUp',
+            { controller: 'UserCtrl' }
+          );
+        });
+      });
+    });
+
+    context('party has 4 members', function() {
+      beforeEach(function() {
+        scope.group = { memberCount: 4 };
+      });
+
+      context('user has "Party Up" but not "Party On" achievement', function() {
+        beforeEach(function() {
+          user.achievements.partyUp = true;
```

Comment:
I don't find doing this kind of setup in a beforeEach very useful when there is only one it block.